File: LeetCode/Sept-2023/896. Monotonic Array/isMonotonic.js
class Solution {
    isMonotonic(nums) {
        const n = nums.length;
        let inc = true;
        let desc = true;

        for (let i = 1; i < n; i++) {
            if (nums[i] - nums[i - 1] < 0) {
                inc = false;
            }
            if (nums[i] - nums[i - 1] > 0) {
                desc = false;
            }
        }

        return inc || desc;
    }
}

File: LeetCode/Sept-2023/896. Monotonic Array/isMonotonic.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        
        int n = nums.length;
        boolean inc = true;
        boolean desc = true;

        for(int i = 1; i<n; i++){
            if(nums[i] - nums[i-1] < 0) inc = false;

            if(nums[i] - nums[i-1] > 0) desc = false;
        }

        return inc || desc;
    }
}